### PR TITLE
Landgrif-137   indicator sources crud

### PR DESCRIPTION
**Pull request by tiagojsag**

### CODE REVIEW

File: api/src/modules/sourcing-record-groups/dto/create.sourcing-record-group.dto.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
```
Comment (alexeh) [inline api/src/modules/sourcing-record-groups/dto/create.sourcing-record-group.dto.ts:1]: File from https://github.com/Vizzuality/landgriffon/pull/46 PR

File: api/src/modules/sourcing-record-groups/dto/update.sourcing-record-group.dto.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { ApiProperty, PartialType } from '@nestjs/swagger';
```
Comment (alexeh) [inline api/src/modules/sourcing-record-groups/dto/update.sourcing-record-group.dto.ts:1]: File from https://github.com/Vizzuality/landgriffon/pull/46 PR

File: api/src/modules/sourcing-record-groups/sourcing-record-group.entity.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import {
```
Comment (alexeh) [inline api/src/modules/sourcing-record-groups/sourcing-record-group.entity.ts:1]: File from https://github.com/Vizzuality/landgriffon/pull/46 PR

File: api/src/modules/sourcing-record-groups/sourcing-record-group.repository.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { EntityRepository, Repository } from 'typeorm';
```
Comment (alexeh) [inline api/src/modules/sourcing-record-groups/sourcing-record-group.repository.ts:1]: File from https://github.com/Vizzuality/landgriffon/pull/46 PR

File: api/src/modules/sourcing-record-groups/sourcing-record-groups.controller.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
+import {
```
Comment (alexeh) [inline api/src/modules/sourcing-record-groups/sourcing-record-groups.controller.ts:1]: File from https://github.com/Vizzuality/landgriffon/pull/46 PR

File: api/src/modules/sourcing-record-groups/sourcing-record-groups.entity.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import {
```
Comment (alexeh) [inline api/src/modules/sourcing-record-groups/sourcing-record-groups.entity.ts:1]: File from https://github.com/Vizzuality/landgriffon/pull/46 PR (**And duplicated entity?**)

File: api/src/modules/sourcing-record-groups/sourcing-record-groups.service.ts [hunk 7]
```diff
@@ -0,0 +1,58 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
```
Comment (alexeh) [inline api/src/modules/sourcing-record-groups/sourcing-record-groups.service.ts:1]: File from https://github.com/Vizzuality/landgriffon/pull/46 PR

File: api/test/sourcing-record-groups/sourcing-record-groups.spec.ts [hunk 8]
```diff
@@ -0,0 +1,133 @@
+import { Test, TestingModule } from '@nestjs/testing';
```
Comment (alexeh) [inline api/test/sourcing-record-groups/sourcing-record-groups.spec.ts:1]: File from https://github.com/Vizzuality/landgriffon/pull/46 PR

File: api/src/modules/sourcing-record-groups/sourcing-record-group.entity.ts [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+  @ApiProperty()
+  lastEdited!: string;
+
+  @ManyToOne(() => User, (user: User) => user.sourcingRecordGroups, {
+    eager: false,
+  })
+  @ApiProperty()
+  lastEditedUser?: User;
+
+  /**
```
Comment (alexeh) [inline api/src/modules/sourcing-record-groups/sourcing-record-group.entity.ts:52]: Causing API to crash, this time not sure which decorator you want to let here - Either `line 46 `or `line 49`
Take in account that making:

```suggestion
  @ApiProperty({type: ()=> User})
```
Will document that the API will retrieve full user info when retrieving a Sourcing Record. I believe this does not reflect the reality

File: api/test/indicator-sources/indicator-sources.spec.ts [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+          title: 'test indicator source',
+          layerId: layer.id,
+        })
+        .expect(HttpStatus.CREATED);
+
+      const createdIndicatorSource = await indicatorSourceRepository.findOne(
+        response.body.data.id,
+      );
+
+      if (!createdIndicatorSource) {
```
[the remark this answers is not in the document]
Comment (tiagojsag) [reply api/test/indicator-sources/indicator-sources.spec.ts:63]: TS complains that `createdIndicatorSource` may be undefined at line 67

[the remark this answers is not in the document]
Comment (alexeh) [reply api/test/indicator-sources/indicator-sources.spec.ts:63]: Oh i see, thanks for clarifying

**Pull request merged**